### PR TITLE
Update GetRecords in AggregationProcess to return map format records

### PR DESCRIPTION
**Pull request by yanjunz97**
This commit updates the return type of GetRecords in AggregationProcess from string to map, which makes it more easier to be structured.

Signed-off-by: Yanjun Zhou <zhouya@vmware.com>

### CODE REVIEW

**Comment by yanjunz97** [comment]
As Antonin suggests, the record should be able to be encoded to JSON format. I changed the return type of the `GetRecords` function to a map of interface. Currently the `GetString()` function in Record is not used. Shall we still keep it?

**Comment by srikartati** [comment]
> As Antonin suggests, the record should be able to be encoded to JSON format. I changed the return type of the `GetRecords` function to a map of interface. Currently the `GetString()` function in Record is not used. Shall we still keep it?

I think JSON format is good right. We can leave it to the user to change in the format they want. If we need some other format, we can add it later. Lets remove for now?

**Comment by yanjunz97** [comment]
> > As Antonin suggests, the record should be able to be encoded to JSON format. I changed the return type of the `GetRecords` function to a map of interface. Currently the `GetString()` function in Record is not used. Shall we still keep it?
> 
> I think JSON format is good right. We can leave it to the user to change in the format they want. If we need some other format, we can add it later. Lets remove for now?

Removed.

File: pkg/intermediate/aggregate.go [hunk 1]
```diff
@@ -215,19 +215,19 @@ func (a *AggregationProcess) GetExpiryFromExpirePriorityQueue() time.Duration {
 	return a.inactiveExpiryTimeout
 }
 
-// GetRecords returns the string format flow records by FlowKey from flowKeyRecordMap.
-// Returns partial match if FlowKey is not complete.
-// Returns all the flow records if FlowKey is not provided.
-func (a *AggregationProcess) GetRecords(flowKey *FlowKey) []string {
+// GetRecords returns map format flow records given a flow key.
```
Comment (srikartati) [inline pkg/intermediate/aggregate.go:218]: Let's explain the map format clearly, where key is element name and value is the IE object.

File: pkg/intermediate/aggregate.go [hunk 2]
```diff
@@ -215,19 +215,19 @@ func (a *AggregationProcess) GetExpiryFromExpirePriorityQueue() time.Duration {
...
+// Returns partial matched flow records if the flow key is not complete.
```
Comment (srikartati) [inline pkg/intermediate/aggregate.go:219]: "partially matched"

Comment (yanjunz97) [reply pkg/intermediate/aggregate.go]: Done

File: pkg/intermediate/aggregate.go (hunk 1 above)
Comment (yanjunz97) [reply pkg/intermediate/aggregate.go:218]: Done

**Review by srikartati — APPROVED** [review]
lgtm

**Review by dreamtalen — APPROVED** [review]
LGTM

**Pull request merged**